custom_components/mikrotik_router/mikrotik_controller.py

```diff
@@ -1215,17 +1233,34 @@ def get_system_resource(self):
         )
 
         tmp_uptime = 0
-        tmp = re.split(r"(\d+)[h]", self.data["resource"]["uptime_str"])
+        tmp = re.split(r"(\d+)[s]", self.data["resource"]["uptime_str"])
         if len(tmp) > 1:
             tmp_uptime += int(tmp[1])
+        tmp = re.split(r"(\d+)[m]", self.data["resource"]["uptime_str"])
+        if len(tmp) > 1:
+            tmp_uptime += int(tmp[1]) * 60
+        tmp = re.split(r"(\d+)[h]", self.data["resource"]["uptime_str"])
+        if len(tmp) > 1:
+            tmp_uptime += int(tmp[1]) * 3600
         tmp = re.split(r"(\d+)[d]", self.data["resource"]["uptime_str"])
         if len(tmp) > 1:
-            tmp_uptime += int(tmp[1]) * 24
+            tmp_uptime += int(tmp[1]) * 86400
         tmp = re.split(r"(\d+)[w]", self.data["resource"]["uptime_str"])
         if len(tmp) > 1:
-            tmp_uptime += int(tmp[1]) * 24 * 7
+            tmp_uptime += int(tmp[1]) * 604800
+
+        now = datetime.now().replace(microsecond=0)
+        uptime_tm = datetime.timestamp(now - timedelta(seconds=tmp_uptime))
+        update_uptime = False;
+        if self.data["resource"]["uptime"] is None or self.data["resource"]["uptime"] == 0:
```

Comment:
Maybe this could be better? Not sure if it will work in this case:
`if not self.data["resource"]["uptime"]`

Reply:
I were not sure neither because the first value after a Home Assistant restart is 0. But I have tested it, and it works :):

```
2021-05-14 12:49:32 WARNING (SyncWorker_1) [custom_components.mikrotik_router.mikrotik_controller] Uptime = 0
2021-05-14 12:49:34 WARNING (SyncWorker_6) [custom_components.mikrotik_router.mikrotik_controller] Uptime = 2021-05-13T18:47:53+02:00
```